dali/base/dasds.cpp

```diff
@@ -5690,59 +5692,71 @@ class CStoreHelper : implements IStoreHelper, public CInterface
         return res;
     }
 
-    unsigned saveStoreToFile(const IPropertyTree *root, IFileIO *iFileIOTmpStore, StoreFormat format)
+    IFile *saveStoreToFile(const IPropertyTree *root, StoreFormat format, unsigned &crc)
     {
         const bool isBinary = (format == StoreFormat::BINARY);
         LOG(MCdebugProgress, "Saving store to %s", isBinary ? "binary" : "XML");
         assertex(root);
-        assertex(iFileIOTmpStore);
 
-        // iFileIOTmpStore is passed in so that it can be used by the caller after the saveStoreToBinaryFile() call,
-        // for error processing and rename etc
-        IFile *iFile = iFileIOTmpStore->queryFile();
+        StringBuffer tmpStoreName;
+        Owned<IFileIO> iFileIOTmpStore(createUniqueFile(location, VStringBuffer("dali_store_tmp%s", isBinary ? ".bin" : ".xml").str(), nullptr, tmpStoreName));
+        Owned<IFile> iFile = LINK(iFileIOTmpStore->queryFile());
```

Comment:
future:  more normal is

Linked<IFile> iFile = iFileIOTmpStore->queryFile();

However if this was being done from scratch we wouldn't have Owned and Linked...